OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -828,14 +841,37 @@ void TracerThread::ProcessSampleEvent(const perf_event_header& header,
     thread_name.set_timestamp_ns(event->GetTimestamp());
     listener_->OnThreadName(std::move(thread_name));
 
+  } else if (is_user_instrumented_tracepoint) {
+    auto it = ids_to_tracepoint_info_.find(stream_id);
+
+    if (it == ids_to_tracepoint_info_.end()) {
+      return;
+    }
+
+    auto event = ConsumeGenericTracepointPerfEvent(ring_buffer, header);
+
+    if (event->GetPid() != pid_) {
```

Comment:
Like with the other tracepoints, you probably don't want to filter according to the target pid here. I would guess that the majority of them come with a kernel-related pid (e.g., 0 in case of the scheduler).

Reply:
Yes, you are right. Thank you a lot for pointing that out. I want to collect tracepoint events system wide.